### PR TITLE
v1.0.0

### LINKED ISSUE #59: Support for Python 3.11 on Windows
Python 3.11 on Windows still has problems due to `lxml` like every year/every new Python version.

See https://bugs.launchpad.net/lxml/+bug/1977998

-----------
**Related PRs:**
* #51

### PR DESCRIPTION
**Pull request by Paebbels**
# New Features
* Added support for Python 3.11 and 3.12-alpha.1 (see #59)
* Added support for PyPy 3.7, 3.8 and 3.9.
* Added support for MSYS2 runtimes: msys, mingw32, mingw64, clang32, clang64 and ucrt64.  
  (Before it was MinGW64 only.)
* Prepared static type checking to support HTML and XML reports as well as artifacts.  
  See #41.
* Wrote a full documentation hosted at GitHub Pages: https://pytooling.github.io/Actions/ generated using BuildTheDocs.
  * Mostly transferred README and other Markdown contents to ReST.
  * Documented all job templates.
    * description
    * inputs
    * secrets
    * outputs
    * usage examples
  * Documented action dependencies.
* Added test pipeline(s) to verify job templates.
  * Parameters are verified.
  * ArtifactCleanup is verified.
* Split paths to e.g. unittests into 2 parameters for tests directory (working directory) and the subdirectory for unittests.
* New first-level outputs from `Parameters`:
  * `python_version` - Python version is now a root element and does need to decode `params`.
  * `artifact_names` - The list of artifact names is now a root element and does need to decode `params`. Compared to `params` it contains more artifact names and is prepared to contain more artifacts and artifact variants like XML vs. HTML.
* Providing `ENVIRONMENT_NAME` in unit tests and code coverage collection jobs.
* The unit testing job template extracts Python packages needed to be installed via pacboy automatically by comparing dependencies with a list of pacboy packages.
* Added editorconfig and vscode settings file.
* Added pull-request template.
* Added dependabot configuration file.

# Changes
* Bumped Python version in MinGW64 from 3.9 to 3.10
* Bumped default Python version to 3.11
* Bumped used GitHub Actions to latest versions:
  * `actions/checkout@v2` &rarr; `actions/checkout@v3`
  * `actions/setup-python@v2`&rarr; `actions/setup-python@v4`
  * `actions/download-artifact@v2` &rarr; `actions/download-artifact@v3`
  * `actions/upload-artifact@v2` &rarr; `actions/upload-artifact@v3`
  * `codecov/codecov-action@v1` &rarr; `codecov/codecov-action@v3`
  * `codacy/codacy-coverage-reporter-action@master` &rarr; `codacy/codacy-coverage-reporter-action@v1`
  * `geekyeggo/delete-artifact@v1` &rarr; `geekyeggo/delete-artifact@v2`
  * `actions/upload-artifact@master` &rarr; `actions/upload-artifact@v3`
  * Action `geekyeggo/delete-artifact@v2` now supports globbing.
  * Actions will use Node.js 16 instead of Node.js 12.
* Split `report` and `artifact` input parameters of `StaticTypeCheck` template into a junit and html variant:
  * Renamed `report` &rarr; `html_report`.
  * Renamed `artifact` &rarr; `html_artifact`.
  * Added `junit_report`.
  * Added `junit_artifact`.
* When specifying packages for MSYS2's `pacboy`, only additional packages need to be listet.
* Changed echoing a special `::set-output::` string, now writing to a `$GITHUB_OUTPUT` file.
* Warn if system `msys2` is used and propose to use `mingw64`.
* Emit an error on empty system list.
* Use default python version, if python version list is empty.
* Removed `pip install -U pip` for updating pip to avoid warnings. Using `--disable-pip-version-check` instead. See #50.
* Deprecated output `params` from `Parameters`.

# Bug Fixes
* Fixed ReST directives in documentation license file.
* Fixes Node.js 12 warnings.
* Fixed `::set-output::` warnings.

# Related Issues and PRs

* Fixes #59 
* #68 
* #69 
* #70 
* #71 
* #72

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    commit-message:
+      prefix: "[Dependabot]"
+    labels:
+      - Dependencies
+    assignees:
+      - Paebbels
+      - Umarcor
+    reviewers:
+      - Paebbels
+      - Umarcor
```
Comment (umarcor) [inline .github/dependabot.yml:16]: Please, use `umarcor` or `UMarCor`.

**Review by umarcor — COMMENTED** [review]
LGTM.

File: .github/dependabot.yml (hunk 1 above)
Comment (Paebbels) [reply .github/dependabot.yml]: Fixed in b5f57165.

**Comment by Paebbels** [comment]
@umarcor I know this is a big PR and therefore I would like to go to v1.0.0. Besides some testing workflows in this repository I also tested all changes in pyTooling/pyTooling using `...@dev`.

So far all issues are fixed.

All stuff is now documented at https://pytooling.github.io/Actions/index.html. I wrote a full blown documentation on it.

----
I you have further ideas, it would be a good place to add new stuff or more breaking stuff in this release.

**Comment by Paebbels** [comment]
Argh, I forgot #41.

Should we also bring the example pipeline into the documentation?

**Comment by Paebbels** [comment]
I'll try to find a solution for the `pacboy` lines.

**Comment by Paebbels** [comment]
**Updates:**
* #41 is now integrated.
* Packages for installation through pacboy are now inferred based on the given requirements list or file.
* Moreover, in a workflow instantiation, only additional packages need to be listed in stead of all packages.
* Upload documentation directly if no artifact is defined.

**Comment by Paebbels** [comment]
Python 3.11 on Windows still has problems due to `lxml` like every year/every new Python version. See https://bugs.launchpad.net/lxml/+bug/1977998

**Comment by Paebbels** [comment]
@umarcor  
Does your merge and push mean are we going to integrate this?
I can explain what this big changeset does, if needed.

Almost all our Python projects are nowadays using the `dev` branch.

**Pull request merged**